front/lib/api/assistant/conversation.ts

```diff
@@ -2023,6 +2066,79 @@ async function updateConversationGroups({
   );
 }
 
+/**
+ * Update the conversation requestedGroupIds based on the mentioned agents.
+ * This function is purely additive - requirements are never removed.
+ *
+ * Each agent's requestedGroupIds represents a set of requirements that must be
+ * satisfied. When an agent is mentioned in a conversation, its requirements are
+ * added to the conversation's requirements.
+ *
+ * Within each requirement (sub-array), groups are combined with OR logic.
+ * Different requirements (different sub-arrays) are combined with AND logic.
+ */
+export async function updateConversationRequestedGroupIds(
+  mentionedAgents: LightAgentConfigurationType[],
+  conversation: ConversationType,
+  t: Transaction
+): Promise<void> {
+  const newRequirements = mentionedAgents.flatMap(
+    (agent) => agent.requestedGroupIds
+  );
+  const currentRequirements = conversation.requestedGroupIds;
+
+  // Check if each new requirement already exists in current requirements.
+  const hasNewRequirements = newRequirements.every((newReq) =>
```

Comment:
Are you sure about this one? Looks off

Reply:
If true then all newRequirements follow 
```
currentRequirements.some((currentReq) =>
      isEqual(sortBy(newReq), sortBy(currentReq))
    )
```
    
    Then below
 ```
  const missingRequirements = newRequirements.filter(
    (newReq) =>
      !currentRequirements.some((currentReq) =>
        isEqual(sortBy(newReq), sortBy(currentReq))
      )
  );
  ```
  
  would be empty?

Reply:
Not sure to understand. Explaining the code here:

The code checks if all new requirements are already included in the conversation's existing requirements (using sorted arrays for comparison).
If `hasNewRequirements` is true, it means all new requirements are already present, so we can exit early. Otherwise, we compute the missing ones.

(will rename the variables to make it clearer!)

Anything that I missed?

Reply:
Dropping my tests here, if it helps:

```
>> newRequirements [
  [ 'group1', 'group2' ],
  [ 'group3', 'group4' ],
  [ 'group5', 'group6' ],
  [ 'group7', 'group8' ]
]
>> currentRequirements [ [ 'group1', 'group2' ] ]
>> hasNewRequirements false
>> missingRequirements [
  [ 'group3', 'group4' ],
  [ 'group5', 'group6' ],
  [ 'group7', 'group8' ]
]
```

Reply:
Ok was confused by the name hasNewRequirements 👍 Should be named alreadyHasNewRequirements I think otherwise it's naturally understood the other way around.

Reply:
Went with `areAllRequirementsPresent`.